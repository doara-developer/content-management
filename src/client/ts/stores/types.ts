import { Item } from "@common/types";

export interface RootState {
    version: string;
}

export interface ItemListState {
    itemList: Item[];
}

export const DialogTypeEnum = {
    None: "None",
    RegistrationForm: "RegistrationForm",
    EditForm: "EditForm",
} as const;
export type DialogType = typeof DialogTypeEnum[keyof typeof DialogTypeEnum];
export type EditFormDataType = {
    item: Item;
};
export type DialogDataType = {} | EditFormDataType;

export interface DialogState {
    type: DialogType;
    data: DialogDataType;
}
