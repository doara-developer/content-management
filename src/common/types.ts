export type Item = {
    id: string;
    name: string;
    purchaseDate: string;
};

export type GetItemsResponse = {
    item_list: Item[];
};
