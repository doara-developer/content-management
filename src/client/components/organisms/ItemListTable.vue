<template>
    <div class="item-list-table">
        <div class="item-list-header">
            <div class="item-list-name">名前</div>
            <div class="item-list-date">最終購入日</div>
            <div class="item-list-action">操作</div>
        </div>
        <div
            v-for="item in itemList"
            :key="item.id"
            class="item-list-table-content"
        >
            <div class="item-list-name">{{ item.name }}</div>
            <div class="item-list-date"></div>
            <div class="item-list-action"></div>
        </div>
    </div>
</template>
<script lang="ts">
import Vue from "vue";

export default Vue.extend({
    computed: {
        itemList() {
            return this.$store.state.itemList.itemList;
        },
    },
});
</script>
<style lang="scss" scoped>
.item-list-table {
    width: 80%;
    border-radius: 5px;
    border: 1px solid #ded5c0;
}
.item-list-header {
    @include display-flex($justify: flex-start);
    width: 100%;
    background-color: #efe1be;
    color: #401c00;
    border-radius: 5px 5px 0px 0px;
    padding: 5px;
}
.item-list-table-content {
    @include display-flex($justify: flex-start);
    width: 100%;
    color: #401c00;
    padding: 5px;
    border-bottom: 1px solid #ded5c0;
}
.item-list-name {
    width: 50%;
}
.item-list-date {
    width: 30%;
}
.item-list-action {
    width: 20%;
}
</style>
