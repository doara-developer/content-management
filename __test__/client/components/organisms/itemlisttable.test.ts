import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import { Item } from "@common/types";
import ItemListTable from "@client/components/organisms/ItemListTable.vue";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("ItemListTable.vue", () => {
    let store: any;
    beforeEach(() => {
        const dummyItemList: Item[] = [
            {
                id: "123",
                name: "dummy_item",
                purchaseDate: "2021-05-21",
            },
        ];
        store = new Vuex.Store({
            state: {
                itemList: {
                    itemList: dummyItemList,
                },
            },
        });
    });
    test("renders props.msg when passed", () => {
        const wrapper = shallowMount(ItemListTable, { store, localVue });
        expect(wrapper).toMatchSnapshot();
    });
});
