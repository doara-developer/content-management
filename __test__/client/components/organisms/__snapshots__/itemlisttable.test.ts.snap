// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemListTable.vue renders props.msg when passed 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
